File: frontend/src/services/feedback.ts
import api from "./api";

export const feedbackService = {
  submit(payload: {
    message_id: number;
    rating: "positive" | "negative";
    comment?: string;
  }) {
    return api.post("/feedback", payload);
  },

  getMyFeedback() {
    return api.get("/feedback/my");
  },

  getAllFeedback() {
    return api.get("/feedback/all");
  },
};

File: backend/app/models/feedback.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    ForeignKey,
    DateTime,
)
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.base import Base


class Feedback(Base):
    __tablename__ = "feedback"

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    user_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False,
    )

    message_id = Column(
        Integer,
        ForeignKey("messages.id"),
        nullable=False,
    )

    rating = Column(
        String,
        nullable=False,
    )  # positive / negative

    comment = Column(
        Text,
        nullable=True,
    )

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
    )

    user = relationship(
        "User",
        back_populates="feedbacks",
    )

    message = relationship(
        "Message",
        back_populates="feedbacks",
    )